core/app/models/spree/promotion_rule.rb

```diff
@@ -41,6 +41,22 @@ def to_partial_path
       "spree/admin/promotions/rules/#{model_name.element}"
     end
 
+    # Methods for the new discount system. They all return `true` by default
+    # as promotion rules limit the eligibility of a promotion.
+    # What we're trying to accomplish here is that it's clear when a promotion is
+    # eligible for an order, or a line item, or a shipment.
+    def order_discountable?(_order)
```

Comment:
Does the method `applicable?` still makes sense with your work ?

Reply:
No it does not; but I'm very seriously considering keeping `applicable`? and `eligible?`, and instead splitting rules like `Product` up into an order-specific and a line-item specific part. That way most existing rules would continue functioning, and it would make the migration path way easier.

Reply:
I've rebased this PR to make use of `applicable?` and `eligible?`